lib/super_digit.rb

```diff
@@ -1,15 +1,16 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def super_digit(n)
-   
+  return n if n.digits.length == 1
+  new = n.digits.sum
+  return super_digit(new)
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def refined_super_digit(n, k)
```

Comment:
Because of the previous method's time complexity this solution is O( (nk)^2) in time complexity and O(nk) in space.  

If you used dynamic programming you could make it O(n^2 + k^2)